failing testsuite when building in mock for Fedora/Rawhide

When looking at the [build.log](http://luther.ceplovi.cz/git/python-cryptography-pkg.git/tree/build.log?id=afdfb992686d745dec88c862660ee87e0c906412) of my attempt to package python-cryptography for Fedora/RHEL it crashes with this:

```
=================================== FAILURES ===================================
_______________________ TestOpenSSL.test_ssl_ctx_options _______________________
self = <test_openssl.TestOpenSSL object at 0x7fbbefc2e910>
    def test_ssl_ctx_options(self):
        # Test that we're properly handling 32-bit unsigned on all platforms.
        b = Binding()
        assert b.lib.SSL_OP_ALL > 0
        ctx = b.lib.SSL_CTX_new(b.lib.TLSv1_method())
        ctx = b.ffi.gc(ctx, b.lib.SSL_CTX_free)
        resp = b.lib.SSL_CTX_set_options(ctx, b.lib.SSL_OP_ALL)
>       assert resp == b.lib.SSL_OP_ALL
E       assert 2164263935L == 2147486719
E        +  where 2147486719 = <cffi.vengine_cpy.FFILibrary object at 0x7fbc0ce3fe50>.SSL_OP_ALL
E        +    where <cffi.vengine_cpy.FFILibrary object at 0x7fbc0ce3fe50> = <cryptography.hazmat.bindings.openssl.binding.Binding object at 0x7fbbefc2e990>.lib
tests/hazmat/bindings/test_openssl.py:116: AssertionError
_________________________ TestOpenSSL.test_ssl_options _________________________
self = <test_openssl.TestOpenSSL object at 0x7fbc0e26ae10>
    def test_ssl_options(self):
        # Test that we're properly handling 32-bit unsigned on all platforms.
        b = Binding()
        assert b.lib.SSL_OP_ALL > 0
        ctx = b.lib.SSL_CTX_new(b.lib.TLSv1_method())
        ctx = b.ffi.gc(ctx, b.lib.SSL_CTX_free)
        ssl = b.lib.SSL_new(ctx)
        ssl = b.ffi.gc(ssl, b.lib.SSL_free)
        resp = b.lib.SSL_set_options(ssl, b.lib.SSL_OP_ALL)
>       assert resp == b.lib.SSL_OP_ALL
E       assert 2164263935L == 2147486719
E        +  where 2147486719 = <cffi.vengine_cpy.FFILibrary object at 0x7fbc0ce3fe50>.SSL_OP_ALL
E        +    where <cffi.vengine_cpy.FFILibrary object at 0x7fbc0ce3fe50> = <cryptography.hazmat.bindings.openssl.binding.Binding object at 0x7fbc0e26a090>.lib
tests/hazmat/bindings/test_openssl.py:128: AssertionError
============ 2 failed, 71531 passed, 2427 skipped in 151.54 seconds ============
```

Is it that I do something wrong (most likely I missed to hunt down some dependency of the package), or is there a problem in `cryptography`?

(filed in the Red Hat Bugzilla as https://bugzilla.redhat.com/show_bug.cgi?id=1146724)

--- a/tests/hazmat/bindings/test_openssl.py
+++ b/tests/hazmat/bindings/test_openssl.py
@@ -137,9 +137,11 @@ class TestOpenSSL(object):
         ctx = b.ffi.gc(ctx, b.lib.SSL_CTX_free)
         ssl = b.lib.SSL_new(ctx)
         ssl = b.ffi.gc(ssl, b.lib.SSL_free)
+        current_options = b.lib.SSL_get_mode(ssl)
         resp = b.lib.SSL_set_mode(ssl, b.lib.SSL_OP_ALL)
-        assert resp == b.lib.SSL_OP_ALL
-        assert b.lib.SSL_OP_ALL == b.lib.SSL_get_mode(ssl)
+        expected_options = current_options | b.lib.SSL_OP_ALL
+        assert resp == expected_options
+        assert b.lib.SSL_get_mode(ssl) == expected_options
 
     def test_windows_static_dynamic_libraries(self):
         assert "ssleay32mt" in _get_windows_libraries("static")

